### PR TITLE
EZP-25533: allow newer versions of http-cache-bundle

### PR DESCRIPTION
**Pull request by bdunogier**
> Fixes http://jira.ez.no/browse/EZP-25533
> **Requires** https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/289
> Can be merged to 6.0, 6.1 and 6.2.

FosHttpCache uses a hardcoded user hash for anonymous users to maximize performances. The change made in https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/274 made apparent an inconsistency in user hash generation for anonymous users on eZ Platform: calling the HashGenerator on an anonymous request will generate a hash different from the hardcoded one, since the hash depends on the user's role, by essence configurable.

A [change](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/compare/1.3...bdunogier:anonymous_request_hash-1.3) will be proposed to FosHttpCacheBundle that disables hash verification on anonymous requests. This PR 
introduced `AnonymousRequestMatcher` from FosHttpCacheBundle with our session cookie prefix. 

### Testing
Manual:
- configure symfony to use the internal reverse proxy (can probably be tested with Varnish as well)
- create a `/tmp/SessionInput.json` file with valid credentials (`{"SessionInput":{"login":"admin","password":"publish"}}`)
```
# login and save cookies to tmp
curl -c /tmp/cookies.txt -X POST -v http://php55-vm.ezplatform/api/ezp/v2/user/sessions --data {"SessionInput":{"login":"admin","password":"publish"}} -H 'Content-Type: application/vnd.ez.api.SessionInput+json'

# run an authentified request, check that you get Vary header
curl -b /tmp/cookies.txt -v http://php55-vm.ezplatform/ -o /dev/null

# run an anonymous request, check that you get a Vary header
curl -v http://php55-vm.ezplatform/ -o /dev/null

# run an authentified request, check that you get Vary header, and that it's a hit from the first authentified request
curl -b /tmp/cookies.txt -v http://php55-vm.ezplatform/ -o /dev/null
```

Without the patch, the anon request would invalidate cache, and wouldn't have a vary header.

### TODO
- [ ] Change to use a compiler pass that sets the argument.
- [ ] Get the [FosHttpCacheBundle PR](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/289) merged

### CODE REVIEW

**Comment by lolautruche** [comment]
I'd better go through a compiler pass instead of redefining the whole service (more forward compatible).
Other than that: :+1:

**Comment by bdunogier** [comment]
Thank you Jérôme :-)

Any feedback about the FosHttpCacheBundle change, before I open the PR ?

**Comment by lolautruche** [comment]
> About the pass, you mean that I should change the fos service's argument instead of redefining right ?

Yup

> Any feedback about the FosHttpCacheBundle change, before I open the PR ?

No, looked good at first sight, but I guess we'll comment on your other PR :wink:

**Comment by bdunogier** [comment]
> No, looked good at first sight, but I guess we'll comment on your other PR :wink: 

Yesplease :-)

File: eZ/Bundle/EzPublishCoreBundle/Resources/config/cache.yml [hunk 1]
```diff
@@ -63,6 +63,10 @@ services:
         class: %ezpublish.http_cache.event_dispatcher.class%
         arguments: [@service_container]
 
+    fos_http_cache.user_context.anonymous_request_matcher:
+        class: %fos_http_cache.user_context.anonymous_request_matcher.class%
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/Resources/config/cache.yml]: The parameter wa

**Comment by bdunogier** [comment]
Changes were made to the FOS PR, in which case no customization should be required. Will close and reopen if applicable.

**Pull request closed without merging**